### ISSUE TITLE
Rule Request: [empty_catch]

### ISSUE THREAD

**Issue by 72A12F4E** [realm/SwiftLint#5634]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### New rule request

Please describe the rule idea, format
this issue's title as `Rule Request: [Rule Name]` and describe:

1. Why should this rule be added? Share links to existing discussion about what
   the community thinks about this.
   
I think we should have a rule that triggers when someone writes a `do-catch`, and leaves the `catch` empty. Cases like this should just use `try?`
   
2. Provide several examples of what _would_ and _wouldn't_ trigger violations.

**Bad**
```swift
do {
    try doStuff()
} catch {

}
```

**Good**
```swift
do {
    try doStuff()
} catch {
   Logger().error("\(error.localizedDescription)")
}
```

3. Should the rule be configurable, if so what parameters should be configurable?

no configuration needed

4. Should the rule be opt-in or enabled by default? Why?
   See [README.md](../README.md#opt-in-rules) for guidelines on when to mark a rule as opt-in.

I think this should be enabled by default, although I can understand the amount of noise that might cause for existing projects.

**Comment by SimplyDanny** [comment]
This could become a part of the new rule introduced in #5617 or a dedicate rule.
